state/state.go

```diff
@@ -538,8 +538,8 @@ func (st *State) MongoSession() *mgo.Session {
 	return st.session
 }
 
-func (st *State) Watch() *Multiwatcher {
-	return NewMultiwatcher(st.workers.allManager())
+func (st *State) Watch(perm permission.Access) *Multiwatcher {
```

Comment:
It seems really inappropriate to be passing in permissions here. State does not do permission checking, so this should be saying more specifically what the scope of the watch is. i.e. whether it should include application offers. Perhaps define a new struct which controls what things to watch?